clang/unittests/Analysis/FlowSensitive/TypeErasedDataflowAnalysisTest.cpp

```diff
@@ -805,6 +805,25 @@ class NullPointerAnalysis final
     else
       JoinedVal.setProperty("is_null", JoinedEnv.makeTopBoolValue());
   }
+
+  std::optional<WidenResult> widen(QualType Type, Value &Prev,
+                                   const Environment &PrevEnv, Value &Current,
+                                   Environment &CurrentEnv) override {
```

Comment:
This appears to be more than an API change in that this model didn't have a widening operation at all before. Can you explain why this has been added? (Is it necessary as part of this patch, or just an additional embellishment?)

Reply:
I added this so that we can test the widening functionality. Turns out we had no tests for framework calls to `widen`.

Reply:
Wow -- makes sense to add this then!